pkg/cmd/server/server.go

```diff
@@ -622,80 +654,122 @@ func (s *server) runControllers(defaultVolumeSnapshotLocations map[string]string
+			s.resticManager,
+		)
+		wg.Add(1)
+		go func() {
+			// TODO only having a single worker may be an issue since maintenance
+			// can take a long time.
+			resticRepoController.Run(ctx, 1)
+			wg.Done()
+		}()
+	}
 
-	resticRepoController := controller.NewResticRepositoryController(
-		s.logger,
-		s.sharedInformerFactory.Velero().V1().ResticRepositories(),
-		s.veleroClient.VeleroV1(),
-		s.sharedInformerFactory.Velero().V1().BackupStorageLocations(),
-		s.resticManager,
-	)
-	wg.Add(1)
-	go func() {
-		// TODO only having a single worker may be an issue since maintenance
-		// can take a long time.
-		resticRepoController.Run(ctx, 1)
-		wg.Done()
-	}()
+	downloadrequestControllerStartup := func() {
+		downloadRequestController := controller.NewDownloadRequestController(
+			s.veleroClient.VeleroV1(),
+			s.sharedInformerFactory.Velero().V1().DownloadRequests(),
+			s.sharedInformerFactory.Velero().V1().Restores(),
+			s.sharedInformerFactory.Velero().V1().BackupStorageLocations(),
+			s.sharedInformerFactory.Velero().V1().Backups(),
+			newPluginManager,
+			s.logger,
+		)
+		wg.Add(1)
+		go func() {
+			downloadRequestController.Run(ctx, 1)
+			wg.Done()
+		}()
+	}
 
-	serverStatusRequestController := controller.NewServerStatusRequestController(
-		s.logger,
-		s.veleroClient.VeleroV1(),
-		s.sharedInformerFactory.Velero().V1().ServerStatusRequests(),
-	)
-	wg.Add(1)
-	go func() {
-		serverStatusRequestController.Run(ctx, 1)
-		wg.Done()
-	}()
+	serverStatusRequestControllerStartup := func() {
+		serverStatusRequestController := controller.NewServerStatusRequestController(
+			s.logger,
+			s.veleroClient.VeleroV1(),
+			s.sharedInformerFactory.Velero().V1().ServerStatusRequests(),
+		)
+		wg.Add(1)
+		go func() {
+			serverStatusRequestController.Run(ctx, 1)
+			wg.Done()
+		}()
+	}
+
+	enabledControllers := map[string]func(){
+		BackupSyncControlleKey:           backupSyncControllerStartup,
+		BackupControllerKey:              backupControllerStartup,
+		ScheduleControllerKey:            scheduleControllerStartup,
+		GcControllerKey:                  gcControllerStartup,
+		BackupDeletionControllerKey:      deletionControllerStartup,
+		RestoreControllerKey:             restoreControllerStartup,
+		ResticRepoControllerKey:          resticRepoControllerStartup,
+		DownloadRequestControllerKey:     downloadrequestControllerStartup,
+		ServerStatusRequestControllerKey: serverStatusRequestControllerStartup,
+	}
+
+	if s.config.restoreOnly {
+		s.logger.Info("Restore only mode - not starting the backup, schedule, delete-backup, or GC controllers")
+		s.config.disabledControllers = append(s.config.disabledControllers,
+			BackupControllerKey,
+			ScheduleControllerKey,
+			GcControllerKey,
+			BackupDeletionControllerKey,
+		)
+	}
+
+	// remove disabled controllers
+	for _, controllerName := range s.config.disabledControllers {
+		_, ok := enabledControllers[controllerName]
+		if ok {
+			delete(enabledControllers, controllerName)
```

Comment:
I think an info log message is warranted as we disable controllers; it would ease troubleshooting in case someone accidentally disabled something.